util/opentitantool_test_runner.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+#
+# A shell script for executing opentitantool as the test harness for
+# functional tests.
+#
+# Currently this script expects only to execute tests via verilator.
+
+set -e
+
+PASS="PASS"
+FAIL="FAIL"
+TIMELIMIT=$((15*60))
+
+for arg in "$@"; do
+  case $arg in
+    --tool=*)
+      OPENTITANTOOL="${arg#*=}"
+      shift
+      ;;
+    --verilator-dir=*)
+      VERILATOR_DIR="${arg#*=}"
+      shift
+      ;;
+    --verilator-rom=*)
+      VERILATOR_ROM="${arg#*=}"
+      shift
+      ;;
+    --verilator-flash=*)
+      VERILATOR_FLASH="${arg#*=}"
+      shift
+      ;;
+    --verilator-otp=*)
+      VERILATOR_OTP="${arg#*=}"
+      shift
+      ;;
+    --pass=*)
+      PASS="${arg#*=}"
+      shift
+      ;;
+    --fail=*)
+      FAIL="${arg#*=}"
+      shift
+      ;;
+    --timeout=*)
+      TIMELIMIT="${arg#*=}"
+      shift
+      ;;
+    *)
+      echo "Unknown argument: $arg"
+      exit 1
+      ;;
+esac
+done
+
+RUST_BACKTRACE=1 ${OPENTITANTOOL} \
```

Comment:
This is going to change somewhat when I add support for dispatch to the cw310 board (see this script in https://github.com/lowRISC/opentitan/pull/9410).

`VERILATOR_FLASH` is going to become `TEST_BIN` and there will be different sets of args passed to each backend.

(Honestly, I'd be fine with squashing 94acedbb1b3fcc8302713ccce66d73ce85e8980d into this PR, but I wasn't sure how things were going to take shape, so initially I wanted to keep #9410 separate)

Reply:
Hmm, I'm still not a big fan of how this is put together. I don't like that the script contains additional flag logic separate from the bazel rule. I think it would make more sense for the script to be a dumb forward to ottool and for all the flags to be set up in the bzl file, so it all lives in one place.

Reply:
I've refactored all of the command-line construction into `rules/opentitan.bzl`.